src/io/data_buffers/distributed_io_buffer.cpp

```diff
@@ -72,7 +72,7 @@ void lbann::distributed_io_buffer::distribute_from_local_matrix(AbsDistMat& Ms,
           << " :: lbann_distributed_io_buffer: No valid data for this step -- local data was invalid";
       lbann_exception(err.str());
     }
-    CopyFromRoot(buf->M_local, buf->Ms);
+    CopyFromRoot(buf->M_local(El::ALL, El::IR(0, Ms.Width())), buf->Ms);
```

Comment:
How does adding the explicit arguments help? Does it avoid a bounds check?

Reply:
`CopyFromRoot` resizes `buf->Ms` to have the same dimensions as `buf->M_local`. This caused the activations matrix to have the wrong dimensions at the last iteration in an epoch. A better approach would be to resize `buf->M_local` to match the current mini-batch step, but this patch was less intrusive.